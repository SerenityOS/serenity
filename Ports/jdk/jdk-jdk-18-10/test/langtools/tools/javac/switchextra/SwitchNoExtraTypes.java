/*
 * @test /nodynamiccopyright/
 * @bug 8206986
 * @summary Verify switch over boolean/long/float/double is not allowed.
 * @compile/fail/ref=SwitchNoExtraTypes.out -XDrawDiagnostics SwitchNoExtraTypes.java
 */

public class SwitchNoExtraTypes {

    private void switchBoolean(boolean b) {
        switch (b) {
            case true: return ;
        }
    }

    private void switchLong(long l) {
        switch (l) {
            case 0: return ;
        }
    }

    private void switchFloat(float f) {
        switch (f) {
            case 0: return ;
        }
    }

    private void switchDouble(double d) {
        switch (d) {
            case 0: return ;
        }
    }

}
