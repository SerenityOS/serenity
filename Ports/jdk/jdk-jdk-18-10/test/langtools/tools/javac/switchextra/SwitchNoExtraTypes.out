SwitchNoExtraTypes.java:11:16: compiler.err.preview.feature.disabled.plural: (compiler.misc.feature.pattern.switch)
SwitchNoExtraTypes.java:12:18: compiler.err.constant.label.not.compatible: boolean, boolean
SwitchNoExtraTypes.java:18:18: compiler.err.constant.label.not.compatible: int, long
SwitchNoExtraTypes.java:24:18: compiler.err.constant.label.not.compatible: int, float
SwitchNoExtraTypes.java:30:18: compiler.err.constant.label.not.compatible: int, double
5 errors
